src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -121,6 +124,35 @@ public void parseEmail_validValueWithWhitespace_returnsTrimmedEmail() throws Exc
         assertEquals(expectedEmail, ParserUtil.parseEmail(emailWithWhitespace));
     }
 
+    @Test
+    public void parseDate_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseDate((String) null));
+    }
+
+    @Test
+    public void parseDate_invalidValue_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseDate(INVALID_DATE));
+    }
+
+    @Test
+    public void parseDate_validValueWithoutWhitespace_returnsDate() throws Exception {
+        DueDate expectedDueDate = new DueDate(VALID_DATE);
+        assertEquals(expectedDueDate, ParserUtil.parseDate(VALID_DATE));
+    }
+
+    @Test
+    public void parseDate_validValueWithoutWhitespace_returnsDateAndTime() throws Exception {
+        DueDate expectedDueDate = new DueDate(VALID_DATE);
+        assertEquals(expectedDueDate, ParserUtil.parseDate(VALID_DATE));
```

Comment:
Maybe have a VALID_TIME = "1200" in you global variable, and for line 145 include both VALID_DATE and VALID_TIME.
Also, in the test method create a variable VALID_DATE_AND_TIME = date + , + time, and use that variable in parseDate.
So it tests when it takes in both date and time

Reply:
oh thx, silly mistake